### ISSUE TITLE
Python error on new project with old files

### ISSUE THREAD

**Issue by Antoviscomi** [3liz/lizmap-plugin#572]
### What is the bug?

1. copy/past features from an old project to an empty new one
2. flag to publish from QGIS SERVER properties 
3. open Lizmap plugin to configure your project

```
`AttributeError: 'NoneType' object has no attribute 'name' 
Traceback (most recent call last):
  File "/home/antonio/.local/share/QGIS/QGIS3/profiles/default/python/plugins/lizmap/plugin.py", line 4447, in run
    self.read_cfg_file()
  File "/home/antonio/.local/share/QGIS/QGIS3/profiles/default/python/plugins/lizmap/plugin.py", line 1827, in read_cfg_file
    data = self.populate_layer_tree()
  File "/home/antonio/.local/share/QGIS/QGIS3/profiles/default/python/plugins/lizmap/plugin.py", line 2216, in populate_layer_tree
    self.process_node(root, None, json_layers)
  File "/home/antonio/.local/share/QGIS/QGIS3/profiles/default/python/plugins/lizmap/plugin.py", line 2179, in process_node
    self.process_node(child, item, json_layers)
  File "/home/antonio/.local/share/QGIS/QGIS3/profiles/default/python/plugins/lizmap/plugin.py", line 2128, in process_node
    self.set_tree_item_data('layer', child_id, json_layers)
  File "/home/antonio/.local/share/QGIS/QGIS3/profiles/default/python/plugins/lizmap/plugin.py", line 2001, in set_tree_item_data
    self.myDic[item_key]['name'] = layer.name()
AttributeError: 'NoneType' object has no attribute 'name''
```

all the layers are (data are from [(https://github.com/3liz/lizmap-web-client/issues/4026#top)]

[files.zip](https://github.com/3liz/lizmap-plugin/files/14618246/files.zip)

Furthermore, 
from LWC in Repository page, I read this in jelix debug console:

![Schermata del 2024-03-15 16-43-49](https://github.com/3liz/lizmap-plugin/assets/12639444/c489f17e-5cf5-41e5-9f2c-087a73d5785e)

and the original project I reshared doesn't are visible in repository but they work fine in QGIS desktop
### Versions, safeguards, checks summary etc

**Versions :**

* Lizmap Web Client : 3.6.10
* Lizmap plugin : 4.2.7
* QGIS Desktop : 3.36.0
* QGIS Server : 3.36.0
* Py-QGIS-Server : not used
* QGIS Server plugin DataPlotly : 4.1.0
* QGIS Server plugin atlasprint : 3.3.2
* QGIS Server plugin lizmap_server : 2.8.5
* QGIS Server plugin wfsOutputExtension : 1.8.0

<details>
<summary>List of safeguards :</summary>
<br/>

* Mode : normal
* Allow parent folder : yes
* Number of parent : 10 folder(s)
* Prevent other drive : no
* Prevent PG service : yes
* Prevent PG Auth DB : yes
* Force PG user&pass : yes
* Prevent ECW : no
</details>

### Check Lizmap plugin

- [X] I have done the step just before in the Lizmap QGIS desktop plugin before opening this ticket. Otherwise, my ticket is not considered valid and might get closed.

**Comment by Antoviscomi** [comment]
In addition, 
When I try to configure by Lizmap Plugin on mine old original project the error message doesn't come out, but the project disappear from repository in LWC (both in 3.6.10 and 3.7.5)

**Comment by Antoviscomi** [comment]
@Gustry I'm just analize the layers/group, deleting one by one, finally whem remain only the layer '_land_use_' the  project works again and the plugin too

**Comment by Gustry** [comment]
> so I think there are something wrong in files of original project (which worked fine until 4.2.7 plugin)

It's not linked to the plugin @Antoviscomi 

QGIS itself is confused with these layers ...

**Comment by Antoviscomi** [comment]
@Gustry that's what I did, but until I removed all layers, left the land_use only, the Python error still appair

**Comment by Antoviscomi** [comment]
@Gustry so I close this topic

**Comment by Antoviscomi** [comment]
> This is very strange ...
> 
> I never saw some layers like this :
> 
> ![image](https://private-user-images.githubusercontent.com/1609292/313765802-319e6ef7-480a-4f22-97bc-6eb04309adbe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA3ODQ1NjEsIm5iZiI6MTcxMDc4NDI2MSwicGF0aCI6Ii8xNjA5MjkyLzMxMzc2NTgwMi0zMTllNmVmNy00ODBhLTRmMjItOTdiYy02ZWIwNDMwOWFkYmUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMThUMTc1MTAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDUxNDNhMWEzN2IwMjRkMjdjY2YwNjliYjkxNzYxZTkwNjZmOWEwZmFmOTE0MDkxODFiYzdmYzEzODJkYjI2MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.rdOUfXKNpWu06mRMMyGY2ULVMEGBFwtpoc8KM7gPjhQ)
> 
> Usually, all layers in QGIS have a an icon the left side, like my green bar on the left.
> 
> But you have a lot of layers in red, they don't have an icon and they don't have a proper menu when we right click.

All layers are from attached gpkg, I copy/paste them because of the error starts with the original project that I didn't edit, and plently worked with previous (Lizmap plugin/LWC) configurations

**Comment by Antoviscomi** [comment]
@Gustry the problem appear with another of my projects. 
I just:
1. open it;
2. check it wit 4.2.7 Lizmap plugin version;
3. automatic solve warnings for project's metadata;
the result is the disappearition of the project ftom LWC index with the error in screenshot below 

![Schermata del 2024-03-20 17-20-40](https://github.com/3liz/lizmap-plugin/assets/12639444/f8cbac29-4e0e-4467-9b10-cc3824593cfd)

(note that my original project name was 'Cetraro2' and not 'CETRARO2' ! Who told to LWC to find a project with total upper case name? )

then I'm not be able to see this project in index page as to public my page on LWC anymore

**Comment by Antoviscomi** [comment]
this is the project restored from my backup

![Schermata del 2024-03-20 18-26-36](https://github.com/3liz/lizmap-plugin/assets/12639444/cc46a836-9a27-4a5c-a4f3-1af40ef5de49)

![Schermata del 2024-03-20 18-32-13](https://github.com/3liz/lizmap-plugin/assets/12639444/bf10f394-6651-43b8-a10b-b98f40bee93f)

when I try to edit the configurations with Lizmap Plugin the project disappear after this:

[Video del 20-03-2024 18:36:49.webm](https://github.com/3liz/lizmap-plugin/assets/12639444/0eaa2a40-755d-453f-addb-c33a4b66fa17)


giving back the error in previuos comment

**Comment by Antoviscomi** [comment]
> Log in your admin panel, and check your logs, I'm sure you will have something like "invalid CFG file", which might be from #571 which is from QGIS 3.36. Please check the content of the CFG file when it has dispeared

no, @Gustry .cfg it's in  correct position and this the log
[LWC_log..txt](https://github.com/3liz/lizmap-plugin/files/14671055/LWC_log.txt)

**Comment by Antoviscomi** [comment]
@Gustry I think I've indexed the trouble:
in cfg file there was some NaN values instead of 0 (zeroes).
I try to explain:
In QGIS desktop, I filter by name a postgis layer named '_limiti_comunali_' (containing Municipality Boudaries), in ordet to public only the boundari ov interest, so in the layer properties the extension become like screenshot below:

![Schermata del 2024-03-21 09-57-07](https://github.com/3liz/lizmap-plugin/assets/12639444/4abcc617-3a4e-4976-a993-469d18e1d629)

and the Lizmap plugin translate these values of extension as NaN (but zero is not Not a Number ;-) ! ) so in cfg I can read this: 

![Schermata del 2024-03-21 09-56-15](https://github.com/3liz/lizmap-plugin/assets/12639444/e1773237-6947-46cc-b5af-6beccbd4dd4d)

well, when I manually turn these values to 'zeroes' the project magically reappear in repository and also works as expected in LWC.

 Having said this, I can't say if the error is in the Lizmap plugin or in QGIS, let me know how to proceed to avoid this boring workaround while waiting for updates

**Comment by Gustry** [comment]
@Antoviscomi As I said yesterday : 

> which might be from https://github.com/3liz/lizmap-plugin/issues/571 which is from QGIS 3.36. Please

**Comment by Gustry** [comment]
I keep this ticket open, but it's a different issue with this screenshot : 

![image](https://github.com/3liz/lizmap-plugin/assets/1609292/81fa7d41-2d51-4244-99dc-d0a571eb4e8b)

**Comment by Antoviscomi** [comment]
> @Antoviscomi As I said yesterday :
> 
> > which might be from #571 which is from QGIS 3.36. Please

@Gustry yes, of course, but, with zeroes as in QGIS and not '_NaN_' (as Lizmap plugin it seems to read from) the issue doesn' exist!

**Comment by Gustry** [comment]
Yes #571, it's `Nan` values in CFG, due to QGIS 3.36 with layers from PG

**Comment by Antoviscomi** [comment]
@Gustry all the layer in my project are from PG, why only this came out with NaN?

**Comment by Gustry** [comment]
@Antoviscomi Please check issue #571, it's linked itself to an issue with PG layers on QGIS

**Comment by Antoviscomi** [comment]
@Gustry which Python commands you mean

**Comment by Gustry** [comment]
The one in #571 : https://github.com/3liz/lizmap-plugin/issues/571#issuecomment-1997060838

**Comment by Antoviscomi** [comment]
here are the results
```
iface.activeLayer().extent().isNull()
False
iface.activeLayer().extent().isEmpty()
False
```

**Comment by Antoviscomi** [comment]
@gustry I would like to point out that with the new check added in the 4.3.0 plugin (version), even tables without geometry are considered to have damaged spatial extension, and on layers with extension problems like in [#571](https://github.com/3liz/lizmap-plugin/issues/571), even though acting as the 'Rebuild Spatial Extension' property, this does not happen null and the values are always read as NaN, moreover I adding in [#571](https://github.com/3liz/lizmap-plugin/issues/571) the behaviour in mine installation of 3.36.0 that persists in 3.36.1

**Comment by Antoviscomi** [comment]
@Gustry manually turning NaN to Zeroes make these layers able to be published

**Comment by Gustry** [comment]
> I would like to point out that, with the new check added in the 4.3.0 plugin (version), even tables without geometry are considered to have damaged spatial extension

Sorry, yes maybe. Can you try https://packages.3liz.org/pub/lizmap-qgis-plugin/unstable/lizmap.4.3.2-alpha.zip ?

> @Gustry manually turning NaN to Zeroes make these layers able to be published

I'm not sure if there is some side effect just by setting them to `0,0,0,0`. I was wondering the plugin should to that.

Can you run https://postgis.net/docs/ST_EstimatedExtent.html on your table ?

**Comment by Antoviscomi** [comment]
@gustry 

> Can you run https://postgis.net/docs/ST_EstimatedExtent.html on your table ?

what QGIS DBManager see:

![Schermata del 2024-03-25 15-22-48](https://github.com/3liz/lizmap-plugin/assets/12639444/f3a84121-13ae-48dd-9e08-65de2a230bed)

response of st_estimatedexent query on QGIS DBMananger 

![Schermata del 2024-03-25 15-23-31](https://github.com/3liz/lizmap-plugin/assets/12639444/582c23a6-10d8-4f99-b033-7b1615e83892)

as in PGAdmin

![Schermata del 2024-03-25 15-27-21](https://github.com/3liz/lizmap-plugin/assets/12639444/bc1124b9-aa36-4ef8-8e8d-9523305a4eb5)




> Sorry, yes maybe. Can you try https://packages.3liz.org/pub/lizmap-qgis-plugin/unstable/lizmap.4.3.2-alpha.zip ?

Now it seems to works better, but all my layers are detected as invalid extent layers as in screencast

[Video del 25-03-2024 15:31:11.webm](https://github.com/3liz/lizmap-plugin/assets/12639444/e07fe8d8-d02c-4561-a02d-02c0d0059169)

with the project not available in repository 

![Schermata del 2024-03-25 15-32-11](https://github.com/3liz/lizmap-plugin/assets/12639444/0b6da883-5e0c-4808-a204-3be42ec51b83)

> I'm not sure if there is some side effect just by setting them to `0,0,0,0`. I was wondering the plugin should to that.

in order to workaround on this I act as in below

[Video del 25-03-2024 15:34:04.webm](https://github.com/3liz/lizmap-plugin/assets/12639444/c02d5565-c85d-456f-a401-4544aa3b23ec)

and this is the result its the ability to correctly publish my map included all layers recognized as invalid extension layers

[Video del 25-03-2024 15:38:08.webm](https://github.com/3liz/lizmap-plugin/assets/12639444/f2eefc3b-4e28-404f-a179-831afbf7c845)

I hope I was clear in my explanation

**Comment by Antoviscomi** [comment]
I.e. limiti_comunali local, with extent abnormal
[Video del 25-03-2024 15:57:32.webm](https://github.com/3liz/lizmap-plugin/assets/12639444/a9bb2ced-456b-4cf4-80e5-3f393ba8195b)
after substitution of NaN with zeroes

[Video del 25-03-2024 15:58:08.webm](https://github.com/3liz/lizmap-plugin/assets/12639444/cdb955c8-5708-493a-90b9-9f65985938c2)

**Comment by Gustry** [comment]
You can at least skip the text editor : 08eab555a40a077dcb1d7dae2d0625fe56774b68
Please try the new version 4.3.3

> after substitution of NaN with zeroes

And is-it OK as far as you can see ?

**Comment by Antoviscomi** [comment]
@gustry yes! it works now, but the user needs to click a tons of ok button (one per layer) in order to see the Lizmap plugin tab to disappear

**Comment by Gustry** [comment]
Well, just click "Ok" :) It's saving the CFG file ...

I know, this is a temporary solution, it's to keep it mind that something might get wrong.

**Comment by Antoviscomi** [comment]
Right @Gustry thank you very much

**Comment by Gustry** [comment]
It's like in any software (QGIS Desktop, other softwares) : 

* Apply : save the configuration but keep the dialog open
* OK : save the configuration but close the dialog

**Comment by Antoviscomi** [comment]
@Gustry in this case ok works as apply while all the recognized invalid extent layers (in my projects are 22) are connfigured

**Comment by Gustry** [comment]
Fixed in 3.36.3
